Sources/SwiftyRequest/RestRequest.swift

```diff
@@ -504,11 +504,15 @@ public class RestRequest {
                 return
             }
 
+            // Retrieve string encoding type
+            let encoding = self.getCharacterEncoding(from: response?.allHeaderFields["Content-Type"] as? String)
+    
             // parse data as a string
-            guard let string = String(data: data, encoding: .utf8) else {
+            guard let string = String(data: data, encoding: encoding) else {
                 let result = Result<String>.failure(RestError.serializationError)
                 let dataResponse = RestResponse(request: self.request, response: response, data: nil, result: result)
                 completionHandler(dataResponse)
+                print("failed to decode string")
```

Comment:
Should we be printing here? Is there a logging policy for SwiftyRequest (I don't see `print()` used elsewhere in the package, nor a logger)? I would normally shy away from printing things to the console in a way that cannot be controlled by a user.

Reply:
Nope, that was just for testing and I forgot to take it out. It might be worth opening another issue to add logging throughout the whole repo at some point. At the moment there isn't any.